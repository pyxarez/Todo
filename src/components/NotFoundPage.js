import React from 'react';
import { browserHistory } from 'react-router';

export default function NotFound() {
  const goBack = () => {
    browserHistory.push('/main');
  }

  return (
    <div>
      <h1>Task not found</h1>
      <button onClick={goBack}>Main page</button>
    </div>
  );
}