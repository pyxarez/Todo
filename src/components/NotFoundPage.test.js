import React from 'react';
import { shallow } from 'enzyme';

import NotFoundPage from './NotFoundPage';

jest.mock('react-router', () => {
  return {
    browserHistory: {
      push: jest.fn()
    }
  };
});
import { browserHistory } from 'react-router';

const setup = () => {
  const wrapper = shallow(<NotFoundPage />)

  return {
    wrapper
  }
}

describe('Components :: NotFoundPage', () => {
  it('should render self', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should go to corresponding URL when press button', () => {
    const { wrapper } = setup();

    wrapper.find('button').simulate('click');
    expect(browserHistory.push)
      .toBeCalledWith('/main');
  });
});