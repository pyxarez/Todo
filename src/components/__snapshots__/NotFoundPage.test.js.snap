exports[`Components :: NotFoundPage should render self 1`] = `
<div>
  <h1>
    Task not found
  </h1>
  <button
    onClick={[Function]}>
    Main page
  </button>
</div>
`;
